src/Servers/Kestrel/Core/src/Internal/Infrastructure/Heartbeat.cs

```diff
@@ -24,6 +24,7 @@ public Heartbeat(IHeartbeatHandler[] callbacks, ISystemClock systemClock, IDebug
         _debugger = debugger;
         _trace = trace;
         _interval = Interval;
+        _stopWaitHandle = new ManualResetEvent(false);
```

Comment:
aside, is "Kestrel Timer" the best name? that's pretty generic. "Kestrel Heartbeat" ? feel free to ignore.

Reply:
Yeah, the name isn't great. I would change it but, the name shows up in tooling. Someone might depend on the name being "Kestrel Timer".

Reply:
It does show in tooling, but I've named (and probably renamed) numerous threads in the core libraries that do and nobody seemed to have an issue -- while perhaps you could write a script (around cordbg?) that uses it, not sure people do.

Having said that I don't care either way.